internal/credentials/xds/handshake_info_test.go

```diff
@@ -300,3 +312,102 @@ func TestMatchingSANExists_Success(t *testing.T) {
 func newStringP(s string) *string {
 	return &s
 }
+
+func TestEqual(t *testing.T) {
+	mockProvider1 := &mockCertProvider{id: 1}
+	mockProvider2 := &mockCertProvider{id: 2}
```

Comment:
Can we please avoid declaring variables outside of the `t.Run()` block. Currently the way things stand, these variables are shared by the different sub-tests and that makes them non-hermetic. Just inline these wherever required.